vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc

```diff
@@ -20,60 +20,69 @@ using std::placeholders::_3;
 
 namespace {
 
-bool GetStatisticalVotingWinner(
+// Allocates one "vote" to a publisher. |dart| is a uniform random
+// double in [0,1] "thrown" into the list of publishers to choose a
+// winner. This function encapsulates the deterministic portion of
+// choosing a winning publisher, separated out into a separate
+// function for testing purposes.
+std::string GetStatisticalVotingWinner(
     double dart,
-    const double amount,
-    const std::vector<ledger::type::ContributionPublisher>& list,
-    ledger::contribution::Winners* winners) {
-  DCHECK(winners);
+    double amount,
+    const ledger::type::ContributionPublisherList& list) {
+  std::string publisher_key;
 
   double upper = 0.0;
-  for (auto& item : list) {
-    upper += item.total_amount / amount;
+  for (const auto& item : list) {
+    upper += item->total_amount / amount;
     if (upper < dart) {
       continue;
     }
 
-    auto iter = winners->find(item.publisher_key);
-
-    uint32_t current_value = 0;
-    if (iter != winners->end()) {
-      current_value = winners->at(item.publisher_key);
-      winners->at(item.publisher_key) = current_value + 1;
-    } else {
-      winners->emplace(item.publisher_key, 1);
-    }
-
-    return true;
+    publisher_key = item->publisher_key;
+    break;
   }
 
-  return false;
+  return publisher_key;
 }
 
+// Allocates "votes" to a list of publishers based on
+// attention. |total_votes| is the number of votes to allocate
+// (typically the number of publishers). |list| is the list of
```

Comment:
The `total_votes` will I think be the number of unspent unblinded tokens, but check me on that.

Reply:
Ah yes, you're right! Will fix.